src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -5,17 +5,16 @@ import React, {
   useContext,
   useMemo,
 } from 'react';
-import { transpose } from 'd3-array';
 import { isNumber } from 'lodash-es';
+import ndarray from 'ndarray';
+import unpack from 'ndarray-unpack';
 import { DimensionMapping, Vis } from './models';
-import { HDF5Dataset, HDF5Value } from '../providers/models';
+import { HDF5Dataset } from '../providers/models';
 import { useValue } from '../providers/hooks';
 import { isSimpleShape } from '../providers/utils';
 
 export interface VisProps {
-  values: HDF5Value;
-  rawDims: number[];
-  slicingIndices?: number[];
+  values: ndarray<number>;
```

Comment:
`VisProps` now only exposes an `ndarray` that is already sliced which means `slicingIndices` are no longer necessary. Also the dimensions are contained in `values.shape` meaning that `rawDims` is also superfluous.